### PR TITLE
Add theme support for post formats

### PR DESCRIPTION
**Pull request by carolinan**
**Description**
I have re-opened this PR after the discussion in https://github.com/WordPress/twentytwentyfive/issues/17
If we want the theme to have partial support for post formats in 6.7, it needs to be added to the theme before Beta 1, since it is an enhancement. Then if it does not work well enough, it can be removed before the string freeze.

This PR:
- Registers theme support for the formats that are listed in Figma: gallery, quote, link, audio, video
- Adds the HTML archive template for the "link" post format.

**Testing Instructions**

This can be tested with the latest version of Gutenberg active.

1. Set the permalinks on the test installation to anything but plain which does not seem to work.
2. Create one or more posts and assign the "link" post format.
3. View the archive on the front: Add `type/link/`after your test domain in the browser address bar.
9. Confirm that the query loop shows the correct posts.

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -9,6 +9,17 @@
  * @since Twenty Twenty-Five 1.0
  */
 
+/**
+ * Add theme support for post formats.
+ *
+ * @since Twenty Twenty-Five 1.0
+ * @return void
+ */
+function twentytwentyfive_post_format_setup() {
```
Comment (juanfra) [inline functions.php]: Let's wrap this in an `if ( ! function_exists(...` like the other functions in the file.

File: templates/taxonomy-post_format-post-format-link.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<!-- wp:template-part {"slug":"header","area":"header","tagName":"header"} /-->
```
Comment (juanfra) [inline templates/taxonomy-post_format-post-format-link.html]:
```suggestion
<!-- wp:template-part {"slug":"header"} /-->
```

File: templates/taxonomy-post_format-post-format-link.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+				<!-- wp:query-pagination-next /-->
+			<!-- /wp:query-pagination -->
+		</div>
+		<!-- /wp:group -->
+	</div>
+	<!-- /wp:query -->
+</main>
+<!-- /wp:group -->
+
+<!-- wp:template-part {"slug":"footer","tagName":"footer"} /-->
```
Comment (juanfra) [inline templates/taxonomy-post_format-post-format-link.html]:
```suggestion
<!-- wp:template-part {"slug":"footer"} /-->
```

**Review by juanfra — COMMENTED** [review]
Thanks Carolina, so the idea is to squeeze it in before beta 1 in case we can make it for 6.7. Right?

File: templates/taxonomy-post_format-post-format-link.html (hunk 2 above)
Comment (carolinan) [reply templates/taxonomy-post_format-post-format-link.html]: I don't think is going to be the final template. 🤷‍♀️

**Comment by carolinan** [comment]
> Thanks Carolina, so the idea is to squeeze it in before beta 1 in case we can make it for 6.7. Right?

I do not believe there is enough support or developers available, but, yes.

**Comment by jasmussen** [comment]
Appreciate this PR.

Do we need additional templates or is this ready to land?

**Comment by carolinan** [comment]
There should be archive templates for each supported format.

And possibly for the singles. The good thing is there are Figma designs for the singles.

**Comment by jasmussen** [comment]
I'll see if I can do anything to help here.

**Comment by jasmussen** [comment]
Following up, I created https://github.com/WordPress/twentytwentyfive/pull/441 with a summary of some learnings. 

One path forward I can see is: we register these formats, but we don't add templates for them. And we create a few patterns for the slightly more elaborate formats, Link and Audio. Let me know your thoughts.

**Comment by carolinan** [comment]
- I removed the HTML template
- Wrapped the theme_support in function exists
- Added the image post format - We did not discuss this, but I think it will be requested by users.

**Review by jasmussen — APPROVED** [review]
This feels worth trying, and it will be easier to rewind if we need to, than add at a later time.

**Pull request merged**